Add job to Circle CI to test docker-compose ubuntu-cpp job

raulcd opened the pull request.

kgpai commented on .circleci/config.yml:
```diff
@@ -25,6 +25,7 @@ workflows:
       - format-check
       - header-check
       - linux-benchmarks-basic
+      - docker-compose-ubuntu-cpp
```
Should we just run this nightly instead of every pr ?

kgpai reviewed (approved):
LGTM, Maybe not run it on every PR ?
Will merge if you can fix merge conflicts.

raulcd commented:
There seems to be a test failure unrelated to the PR https://app.circleci.com/pipelines/github/facebookincubator/velox/16508/workflows/02573e2c-65b5-41dd-91c6-0c0acc70d014/jobs/102349:
```
6: /velox/velox/common/file/tests/FileTest.cpp:176: Failure
6: Expected equality of these values:
6:   localFs->list(std::string_view(tempFolder->path))
6:     Which is: { "/tmp/velox_test_6qwkMs/2", "/tmp/velox_test_6qwkMs/1" }
6:   std::vector<std::string>({a, b})
6:     Which is: { "/tmp/velox_test_6qwkMs/1", "/tmp/velox_test_6qwkMs/2" }
6: [  FAILED  ] LocalFile.list (0 ms)
6: [ RUN      ] LocalFile.readFileDestructor
```
I think this is ready to merge now and we can open a ticket to investigate a possible flaky outside of this PR. @kgpai , I've rebased and changed this job to run on a nightly basis.
